lib/cosmos/packets/structure.rb

```diff
@@ -303,6 +303,34 @@ def set_item(item)
       end
     end
 
+    # @param name [String] Name of the item to delete in the items Hash
+    # @return [StructureItem] StructureItem or one of its subclasses
+    def delete_item(name)
+      item = @items[name.upcase]
+      raise ArgumentError, "Unknown item: #{name}" unless item
+
+      # Find the item to delete in the sorted_items array
+      item_index = nil
+      @sorted_items.each_with_index do |sorted_item, index|
+        if sorted_item.name == item.name
+          item_index = index
+          break
+        end
+      end
+      @sorted_items.delete_at(item_index)
+      # Make a deep copy of all the remaining items so we can re-create them
+      saved = Marshal.load(Marshal.dump(@sorted_items))
+      @items = {}
+      @sorted_items = []
+      @defined_length = 0
+      @defined_length_bits = 0
+      @pos_bit_size = 0
+      @neg_bit_size = 0
+      @fixed_size = true
+      @buffer = '' if @buffer
+      saved.each {|item| append(item) }
```

Comment:
This may seem really inefficient but with the way we append items and dynamically calculate the bit offsets I couldn't figure any other way to do it

Reply:
I don't think I intended for deleting an item, to reassign all the bit offsets.    Maybe you just want to delete an item and put another with a different name in the same place.   Just delete the item and leave everything else alone.

Reply:
We already have a HIDDEN keyword for packets. Maybe we should add that for items and parameters. I was expecting this to completely remove the item. In my demo example if you delete without redoing the offsets you'll have a hidden packet blank space. I think it's useful to have both. Should we open a new ticket?

Reply:
IGNORE_ITEM is for ignoring items.   
Deleting items is only for really weird cases where something is generated really close to what you want, but you need to delete one or two items for some reason.   I don't care if it leaves a packet hole, you can fill that with a specific ITEM definition.  Also, you don't know if that packet was originally created with append or with specific bit offsets, so recreating it with appends doesn't always make sense.  I still think it should just simply delete the item and leave a hole.

Reply:
Perhaps reassigned bit offsets should be an option to DELETE_ITEM?